Core/Src/mcp3008.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * MCP3008.c
+ *
+ */
+
+#include "../Inc/mcp3008.h"
+
+/*
+ * Set the MISO, MOSI, SCK and CS
+ * SPI settings:
+ * CPHA = 1 Edge
+ * Prescaler = 8
+ * First bit = MBS first
+ * CPOL = Low
+ */
+void MCP3008_Init(MCP3008_SPI* spi, SPI_HandleTypeDef* hspi, GPIO_TypeDef* CS_PORT, uint16_t CS_PIN){
+  spi->hspi = hspi;
+  spi->CS_PORT = CS_PORT;
+  spi->CS_PIN = CS_PIN;
+}
+
+// Read the channels from 0 to 7
+uint16_t MCP3008_Read_Channel(MCP3008_SPI* spi, uint8_t channel){
+
+	// Declare data that we will send
+	uint8_t pTxData[3] = {0};
+	pTxData[0] = ((0x01 << 7)|		  // start bit
+		     (1 << 6)|			// SGL
+		     ((channel & 0x07) << 3)); 	// channel number

```

Comment:
I think I understand what you trying to here, but the code is a bit off.
![image](https://github.com/UWARG/embedded-bootcamp/assets/106219057/ce922ad8-27c2-4ca1-8760-8d9b3e818cd1)
The first byte of the TX data is `0x80` which you got it correctly by doing the bit manipulation `(0x01 << 7)`
But the information of the SGL and channel are part of the second TX byte from the data sheet, so you should fix that.

Reply:
gotcha